controller/k8s/api.go

```diff
@@ -71,26 +71,79 @@ type API struct {
 	namespace         string
 }
 
-// NewAPI takes a Kubernetes client and returns an initialized API
+// InitializeAPI creates Kubernetes clients and returns an initialized API wrapper.
+func InitializeAPI(kubeConfig string, namespace string, resources ...APIResource) (*API, error) {
+	k8sClient, err := NewClientSet(kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// check for cluster-wide vs. namespace-wide access
+	clusterAccess, err := k8s.ClusterAccess(k8sClient, namespace)
+	if err != nil {
+		return nil, err
+	}
+	restrictToNamespace := ""
+	if !clusterAccess {
+		log.Warnf("Not authorized for cluster-wide access, limiting access to \"%s\" namespace", namespace)
+		restrictToNamespace = namespace
+	}
+
+	// check for need and access to ServiceProfiles
+	var spClient *spclient.Clientset
+	idxSP := 0
+	needSP := false
+	for i := range resources {
+		if resources[i] == SP {
+			needSP = true
+			idxSP = i
+			break
+		}
+	}
+	if needSP {
```

Comment:
This works, but I would also be okay with removing this logic and just saying that the service profile CRD and permissions are *required*.  That puts us in an intermediate state where users without cluster access can't install Linkerd without help from someone who does have cluster access to install the CRD, but that should be resolved once we have the install flow broken in two stages.

TIOLI

Reply:
Per the [product org's requirement 3](https://github.com/linkerd/linkerd2/issues/2164#issuecomment-458265137):
```
3. Linkerd should still be usable in a locked down environment without elevated permissions, even if that mode has fewer features.
```